probe-rs/src/debug/debug_info.rs

```diff
@@ -704,6 +706,47 @@ impl DebugInfo {
             .get_program_counter()
             .and_then(|pc| pc.value)
         {
+            // PART 0: The first step is to determine the exception context for the current PC.
+            // - If we are at an exception hanlder frame, we need to overwrite the unwind registers with the exception context.
+            // - If for some reason we cannot determine the exception context, we silently continue with the rest of the unwind.
+            // At worst, the unwind will be able to unwind the stack to the frame of the most recent exception handler.
+            let frame_name_override = match core.core_type() {
+                CoreType::Armv6m | CoreType::Armv7em | CoreType::Armv7m | CoreType::Armv8m => {
+                    if let Some(Some(lr_register_value)) =
+                        unwind_registers.get_return_address().map(|r| r.value)
+                    {
+                        let lr_address: u32 = lr_register_value.try_into().map_err(|error| {
+                            crate::Error::Other(anyhow::anyhow!(
+                                "UNWIND: Failed to convert LR register value to address: {:?}. Please report this as a bug.",
+                                error
+                            ))
+                        })?;
+                        if lr_address == 0xFF_FF_FF_F9 {
```

Comment:
Do you have some documentation for this? I think it's a bit more complicated than since, because the actual value depends on the exception number, from what I remember.

Reply:
Yeah, mostly working from the Arm Procedure Calling Standard for now. I know there are a couple of other patterns also, but wanted to start with the easy case and make that work first, before I expand :)

Reply:
I would suggest creating a function for this, that makes it more understandable, and it's also easier to document. The function can be very simple for now, but I think it's best to start with a function already from the beginning.

Reply:
@Tiwalun I've captured the other variations of the LR being used as EXC_RETURN for the currently implemented core types.